src/controls/controls.actions.ts

```diff
@@ -268,138 +270,108 @@ import { renderCircleControl, renderSquareControl } from './controls.render';
   }
 
   /**
-   * Detect if the fabric object is flipped on one side.
-   * @param {fabric.Object} target
-   * @return {Boolean} true if one flip, but not two.
+   * @see https://github.com/fabricjs/fabric.js/pull/8380
    */
-  function targetHasOneFlip(target) {
-    return target.flipX !== target.flipY;
-  }
-
-  /**
-   * Utility function to compensate the scale factor when skew is applied on both axes
-   * @private
-   */
-  function compensateScaleForSkew(
-    target,
-    oppositeSkew,
-    scaleToCompensate,
-    axis,
-    reference
+  function skewObject(
+    axis: TAxis,
+    { target, ex, ey, originX, originY, skewX, skewY, flipX, flipY },
+    pointer: Point
   ) {
-    if (target[oppositeSkew] !== 0) {
-      var newDim = target._getTransformedDimensions()[axis];
-      var newValue = (reference / newDim) * target[scaleToCompensate];
-      target.set(scaleToCompensate, newValue);
-    }
+    const { counterAxis, skew: skewKey } = AXIS_KEYS[axis],
+      dimBefore = target._getTransformedDimensions(),
+      offset = pointer.subtract(new Point(ex, ey))[axis],
+      flip = new Point(flipX ? -1 : 1, flipY ? -1 : 1)[axis],
+      skewingStart = new Point(skewX, skewY)[axis],
+      skewingBefore = new Point(target.skewX, target.skewY)[axis],
+      shearingStart = Math.tan(degreesToRadians(skewingStart)),
+      counterOriginFactor = new Point(
+        target.resolveOriginX(originX),
+        target.resolveOriginY(originY)
+      )[counterAxis],
+      offsetFactor = -Math.sign(counterOriginFactor) * flip * 2;
+
+    const shearing =
+      (offset * offsetFactor) / Math.max(dimBefore[counterAxis], 1) +
```

Comment:
This is the line that needs fixing
if `axis === y` it seems `dimBefore[counterAxis]` should be something else.

@luizzappa would love your help with this

Reply:
test it by skewing X to a big value and then start skewing Y
You will see the the skew change is not li

Reply:
fixed 89d2c0b